tests/integration/integration_go_test.go

```diff
@@ -800,60 +800,61 @@ func TestComponentProviderSchemaGo(t *testing.T) {
 
 // TestTracePropagationGo checks that --tracing flag lets golang sub-process to emit traces.
 func TestTracePropagationGo(t *testing.T) {
-
-	// Detect a special trace coming from Go language plugin.
-	isGoListTrace := func(t *appdash.Trace) bool {
-		m := t.Span.Annotations.StringMap()
-
-		isGoCmd := strings.HasSuffix(m["command"], "go") ||
-			strings.HasSuffix(m["command"], "go.exe")
-
-		if m["component"] == "exec.Command" &&
-			m["args"] == "[list -m -json -mod=mod all]" &&
-			isGoCmd {
-			return true
-		}
-
-		return false
-	}
-
-	var foundTrace *appdash.Trace
-
-	// Look for trace mathching `isGoListTrace` in the trace file
-	// and store to `foundTrace`.
-	searchForGoListTrace := func(dir string) error {
-
-		store, err := ReadMemoryStoreFromFile(filepath.Join(dir, "pulumi.trace"))
-		if err != nil {
-			return err
-		}
-
-		foundTrace, err = FindTrace(store, isGoListTrace)
-		if err != nil {
-			return err
-		}
-
-		return nil
-	}
+	dir, err := os.MkdirTemp("", "pulumi-tracefiles")
+	assert.NoError(t, err)
+	defer os.RemoveAll(dir)
 
 	opts := &integration.ProgramTestOptions{
 		Dir:                    filepath.Join("empty", "go"),
 		Dependencies:           []string{"github.com/pulumi/pulumi/sdk/v3"},
 		SkipRefresh:            true,
-		SkipPreview:            false,
-		SkipUpdate:             true,
+		SkipPreview:            true,
+		SkipUpdate:             false,
```

Comment:
The doubling up of stack exports shows up only in update, so need to switch this test from update to preview.